conda_anaconda_tos/api.py

```diff
@@ -325,3 +334,251 @@ def clean_tos(tos_root: str | os.PathLike[str] | Path) -> Iterator[Path]:
             pass
         else:
             yield path
+
+
+def backup_tos_configs() -> Path:
+    """Backup ToS configuration files from all search path locations.
+
+    Returns:
+        Path to the created backup directory
+
+    Raises:
+        CondaToSBackupError: If backup directory cannot be created or files
+            cannot be copied.
+
+    """
+    backup_timestamp = datetime.now(tz=timezone.utc)
+    timestamped_backup_dir = TOS_BACKUP_DIR / (
+        f"backup_{int(backup_timestamp.timestamp())}"
+    )
+
+    # Create backup directory
+    try:
+        timestamped_backup_dir.mkdir(parents=True, exist_ok=True)
+    except (PermissionError, OSError) as e:
+        raise CondaToSBackupError("create", timestamped_backup_dir, str(e)) from e
+
+    # Collect ToS files with location metadata from all search path locations
+    file_info: list[BackupFileInfo] = []
+
+    for search_path in get_search_path():
+        for tos_file in search_path.glob("*/*.json"):
+            if tos_file.is_file():
+                file_info.append(
+                    BackupFileInfo(
+                        file_path=tos_file,
+                        source_location=str(search_path),
+                        is_temporary=is_temporary_location(search_path),
```

Comment:
This looks like it stores the full path information, so why do we need the priority locations?